tests/integration/__init__.py

```diff
@@ -1091,17 +1091,23 @@ def minion_run(self, _function, *args, **kw):
         return self.run_function(_function, args, **kw)
 
     def run_function(self, function, arg=(), minion_tgt='minion', timeout=25,
-                     **kwargs):
+                     async=False, **kwargs):
         '''
         Run a single salt function and condition the return down to match the
         behavior of the raw function call
         '''
         know_to_return_none = (
             'file.chown', 'file.chgrp', 'ssh.recv_known_host'
         )
-        orig = self.client.cmd(
-            minion_tgt, function, arg, timeout=timeout, kwarg=kwargs
-        )
+        if async:
+            orig = self.client.cmd_async(
+                minion_tgt, function, arg, timeout=timeout, kwarg=kwargs
+            )
+            return orig
+        else:
+            orig = self.client.cmd(
+                minion_tgt, function, arg, timeout=timeout, kwarg=kwargs
+            )
```

Comment:
This is intentional. We don't want to continue processing at this point. We have everything we need. However, the sync version needs to go on for additional checks.